api/user.go

```diff
@@ -151,12 +156,18 @@ func (ua *UserAPI) Post() {
 	user := models.User{}
 	ua.DecodeJSONReq(&user)
 
+	if ua.IsBasicAuth {
+		user.Username = ua.UserNameInBasicAuth
```

Comment:
Sorry but this seems incorrect, username and password in basic auth header is for identifying the caller of the API not the user to be added.